src/components/VueTable.vue

```diff
@@ -504,20 +505,23 @@ export default {
       const realHeightTable = this.$refs[`${this.customTable}-vueTable`].offsetHeight;
       // stock size / offsetTop / offsetLeft of the element
       const width = el.offsetWidth;
-      // stock heightOfScrollbar(40) cell(40) dropdown(140)
-      const heightOfScrollbarCellDropdown = 180;
+      // stock dynamic height of dropdown
+      const dropdownRef = this.$refs[`${this.customTable}-vueTbody`].$refs[`dropdown-${this.customTable}-${colIndex}-${rowIndex}`] || [{}];
+      const heightOfDropdown = (dropdownRef[0] || {}).offsetHeight || 180;
```

Comment:
You already define dropdownRef equal to ```this.$refs[`${this.customTable}-vueTbody`].$refs[`dropdown-${this.customTable}-${colIndex}-${rowIndex}`]``` or `[{}]`

Why do you repeat that here: `dropdownRef[0] || {}` ?

Reply:
That's a bit tricky actually.
`this.$refs[${this.customTable}-vueTbody].$refs[dropdown-${this.customTable}-${colIndex}-${rowIndex}]` can return several results: `undefined`, `[]` or `[<ref component>]`.
1) First case (`undefined`) is solved with the `|| [{}]`. This way, in next line,`dropdownRef[0]` returns the empty object and don't cause an error.
2) Second case (`[]`) is solved with `(dropdownRef[0] || {})`, this way if I have an empty array, I return an empty object instead of undefined.
3) Third case (`[<my component>]`) is the one expected, none of the "or" will be used.

I tested it clicking an all cells to see if the triggered method doesn't lead to an error.

I can edit the first "or" with `$refs[`...`] || [];` meaning `[{}]` becomes `[]`. I would work as well, but we'll used the two undefined solvers (both the "or" || will be used).

Another solution would be to use several `if ( ... !== undefined)` but it's way more verbal and ugly in my opinion :)

Reply:
Ok my bad

Wrap the function with this condition

```
if (this.tbodyData[rowIndex][this.headers[colIndex].headerKey].search === true) {}
```

it return undefined because the renderer of ul is trigger when search equal to true

Reply:
Now adapt your code, the terner is deprecated